webapp/src/components/admin_settings/teams_input/teams_input.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import debounce from 'lodash/debounce';
+import AsyncSelect from 'react-select/async';
+
+// TeamsInput searches and selects teams displayed by display_name.
+// Teams prop can handle the team object or strings directly if the team object is not available.
+// Returns the selected team ids in the `OnChange` value parameter.
+export default class TeamsInput extends React.Component {
+    static propTypes = {
+        placeholder: PropTypes.string,
+        teams: PropTypes.array,
+        onChange: PropTypes.func,
+        actions: PropTypes.shape({
+            searchTeams: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    onChange = (value) => {
+        if (this.props.onChange) {
+            this.props.onChange(value);
+        }
+    }
+
+    getOptionValue = (team) => {
+        if (team.id) {
+            return team.id;
+        }
+
+        return team;
+    };
+
+    formatOptionLabel = (option) => {
+        if (option.display_name) {
+            return (
+                <React.Fragment>
+                    { `${option.display_name}`}
+                </React.Fragment>
+            );
+        }
+
+        return option;
+    }
+
+    debouncedSearchTeams = debounce((term, callback) => {
+        this.props.actions.searchTeams(term).then(({data}) => {
+            callback(data);
+        }).catch(() => {
+            // eslint-disable-next-line no-console
+            console.error('Error searching teams in custom attribute settings dropdown.');
+            callback([]);
+        });
+    }, 150);
+
+    teamsLoader = (term, callback) => {
+        try {
+            this.debouncedSearchTeams(term, callback);
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error(error);
+            callback([]);
+        }
+    };
```

Comment:
I think we can get rid of this function and just pass in the debounced version to the child component. The `debouncedSearchTeams` won't throw an error for an API response error anyway, so this catch block won't run.

Reply:
:+1: